libkbfs/folder_block_ops.go

```diff
@@ -1175,18 +1175,18 @@ func (fbo *folderBlockOps) pathFromNodeForBlockWriteLocked(
 // writeGetFileLocked checks write permissions explicitly for
 // writeDataLocked, truncateLocked etc and returns
 func (fbo *folderBlockOps) writeGetFileLocked(
-	ctx context.Context, lState *lockState, md ReadOnlyRootMetadata,
+	ctx context.Context, lState *lockState, kmd KeyMetadata,
 	file path) (*FileBlock, keybase1.UID, error) {
 	fbo.blockLock.AssertLocked(lState)
 
 	username, uid, err := fbo.config.KBPKI().GetCurrentUserInfo(ctx)
 	if err != nil {
 		return nil, "", err
 	}
-	if !md.GetTlfHandle().IsWriter(uid) {
-		return nil, "", NewWriteAccessError(md.GetTlfHandle(), username)
+	if !kmd.GetTlfHandle().IsWriter(uid) {
```

Comment:
Ah, actually it shouldn't return `nil`; I was just being lazy for `emptyKeyMetadata`. I added comments to the `KeyMetadata` interface, and a comment in `emptyKeyMetadata` explaining why it's returning `nil`. (It shouldn't be used in contexts where the TlfHandle is needed.)